CMakeLists.txt

```diff
@@ -61,12 +61,20 @@ macro(build_libyaml)
   endif()
   include(ExternalProject)
   externalproject_add(libyaml-0.2.5
-    URL https://github.com/yaml/libyaml/archive/refs/tags/0.2.5.zip
-    URL_MD5 2464078985a73f9d298689d36061143f
-    TIMEOUT 60
+    GIT_REPOSITORY https://github.com/yaml/libyaml.git
+    GIT_TAG 2c891fc7a770e8ba2fec34fc6b545c672beb37e6  # 0.2.5
```

Comment:
I'll point out that you can use an actual tag here (like `0.2.5`), which I think would be preferable.

Reply:
I'm not sure that's preferable. Currently a URL to a zip file and an MD5 sum are used. The MD5 sum makes sure the archive contains what we expect it to (limited by the ease of which it is to make an MD5 collision these days). If `GIT_TAG` only says the human readable tag name, then a change to the tag upstream could break the build, or worse be a security issue. Using the sha hash solves that.

Reply:
We discussed this elsewhere, and we are going to switch to git hashes everywhere.  So with that, I'll go ahead and approve.